### ISSUE TITLE
Unify styles of custom operations

### ISSUE THREAD

**Issue by WindQAQ** [tensorflow/addons#156]
Currently, custom operations in addons have no unified styles. For example
the `name` argument in
https://github.com/tensorflow/addons/blob/463112e6e7c8d3623d3d97e6b1f365d1c403f2c4/tensorflow_addons/image/dense_image_warp.py#L161
and
https://github.com/tensorflow/addons/blob/463112e6e7c8d3623d3d97e6b1f365d1c403f2c4/tensorflow_addons/image/transform_ops.py#L38-L42
are not consistent.

As per [coding style for python operation in Tensorflow](https://www.tensorflow.org/community/contribute/code_style#python_operations), the last argument should be `name` with a default value of `None`. Though not all operations in core Tensorflow conform to this (see `tf.image` source code), I'd suppose we should unify the API/style of python operations in addons (like what we do for losses and layers). Maybe we can try to follow the link above first.

**Comment by facaiy** [comment]
Good catch, I think `None` is better for default value, because it's easy to maintain compatibility.

**Comment by WindQAQ** [comment]
Got it, I'll create a PR for image ops first.

**Comment by seanpmorgan** [comment]
I think this is complete? Are there any other ops that need modifying?

**Comment by WindQAQ** [comment]
Just find we miss some:
https://github.com/tensorflow/addons/blob/c2faddd818280815bd5e47b81742efdfc90e7aaf/tensorflow_addons/image/distance_transform.py#L34-L36
https://github.com/tensorflow/addons/blob/c2faddd818280815bd5e47b81742efdfc90e7aaf/tensorflow_addons/image/transform_ops.py#L320

Some ops like `compose_transforms` in [transform_ops.py](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/transform_ops.py) do not have `name` argument (even when they are in contrib though). Would it be better to add `name` to them for API compatibility?

**Comment by facaiy** [comment]
1. For `rotate` op, we should replace `str(name)` by `(name or 'rotate')`. 
2. There exists an known annoying issue about `name`:  if we want to add new attribute, we have to append it after `name` for ensuring backward compatibility.  And all new attributes are forced to be optional. But I agree to conform tensorflow convention: add `name` for all public API in image module. 
 cc @martinwicke Hi, Martin, do you have any good idea?

@seanpmorgan Sean, what do you think?

**Comment by martinwicke** [comment]
The name arg problem is annoying. 

We're not quite ready to remove the name args. At this point, they're exclusively used for visualization in TensorBoard. We have some ideas about visualizing summarized stack traces instead of manually set names, and once that works well, we can think about soft deprecating all name args. 

For now, I'd keep the name args.

**Comment by facaiy** [comment]
Thank @martinwicke for the input. 

@WindQAQ Tzu-Wei, let's add `name` argument for all public operators.
